packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1068,9 +1134,42 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         return this._summarizer;
     }
 
-    private get summariesDisabled(): boolean {
-        return this.runtimeOptions.summaryOptions.disableSummaries === true ||
-            this.runtimeOptions.summaryOptions.summaryConfigOverrides?.disableSummaries === true;
+    public get summariesDisabled(): boolean {
+        if (this.runtimeOptions.summaryOptions.disableSummaries === true) {
+            return true;
+        }
+        return this.summaryConfiguration.state === "disabled";
+    }
+
+    public get heuristicsDisabled(): boolean {
+        if (this.runtimeOptions.summaryOptions.summarizerOptions?.disableHeuristics === true) {
+            return true;
+        }
+        return this.summaryConfiguration.state === "disableHeuristics";
+    }
+
+    public get summarizerClientElectionEnabled(): boolean {
+        if (this.runtimeOptions.summaryOptions.summarizerClientElection === true) {
+            return this.mc.config.getBoolean("Fluid.ContainerRuntime.summarizerClientElection") ?? true;
+        }
+        assert(this.summaryConfiguration.state !== "disabled", "Summary Configuration is invalid");
```

Comment:
Not sure I understand this assert. This config cannot be retrieved if summaries are disabled? That does not sound right.

Reply:
Same comment for `maxOpsSinceLastSummary`

Reply:
maxOpsSinceLastSummary  is only ever used if summaries are enabled or heuristics disabled.  Check Event SummariesDisabled

Reply:
That makes sense. But why assert if someone is trying to get the value?
Also, these should not be getters. The values for these properties should be set when the container runtime is created and they should not change after that. If this is a getter, then the values for these properties can change over the lifetime of the container runtime (by changing feature flags / configs). That should not happen.